packages/ui/components/form/select/SelectDownshift.tsx

```diff
@@ -0,0 +1,248 @@
+import {
+  useCombobox,
+  UseComboboxGetItemPropsOptions,
+  useMultipleSelection,
+  UseMultipleSelectionGetSelectedItemPropsOptions,
+} from "downshift";
+import React, { ReactElement, ReactNode } from "react";
+
+import { classNames } from "@calcom/lib";
+
+import { Divider } from "../../divider";
+import { TextField } from "../inputs/Input";
+import { Label } from "../inputs/Label";
+
+type StringProperties<T> = {
+  [K in keyof T]: T[K] extends string ? K : never;
+}[keyof T];
+
+interface MultiComboboxProps<T extends Record<string, unknown>> {
+  items: T[];
+  /** The field that will be used to display info */
+  labelKey: keyof T;
+  // gets that can be used to search for items in text box
+  searchableKey?: StringProperties<T>;
+  dividerKey?: keyof T;
+  initalSelectedItems?: T[];
+  onSelectedItemsChange?: (selectedItems: T[]) => void;
+  selectNode?: (
+    item: T,
+    index: number,
+    selection: { isActive: boolean; isKeyboardFocused: boolean },
+    // Downshift doesnt supply types here so we have to say as any
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    getItemProps: (options: UseComboboxGetItemPropsOptions<T>) => any
+  ) => ReactNode;
+  renderNode?: (
+    items: T[],
+    // Downshift doesnt supply types here so we have to say as any
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    getSelectedItemProps: (options: UseMultipleSelectionGetSelectedItemPropsOptions<T>) => any,
+    removeSelectedItem: (item: T) => void
+  ) => ReactNode;
+  searchNode?: () => ReactElement<HTMLInputElement>;
+  input?: {
+    className: string; // Nested so intelisense still picks up classNames
+  };
+}
+
+export function MultipleComboBoxExample<T extends Record<string, unknown>>(props: MultiComboboxProps<T>) {
+  const [inputValue, setInputValue] = React.useState("");
+  const [selectedItems, setSelectedItems] = React.useState<T[]>(props.initalSelectedItems || ([] as T[]));
+
+  const items = React.useMemo(() => {
+    const lowerCasedInputValue = inputValue.toLowerCase();
+    return props.items.filter(function filterItem(item) {
+      const itemSearchable = props.searchableKey && (item[props.searchableKey] as string); // we know this is as string due to string properties type
+      return itemSearchable ? itemSearchable?.toLowerCase().includes(lowerCasedInputValue) : props.items;
+    });
+  }, [inputValue, props.items, props.searchableKey]);
```

Comment:
Memo the filter function otherwise it gets a canny few re-renders

TODO: maybe debounce this?